### PR TITLE
Simplify creation of StoreLogService

### PR DESCRIPTION
**Pull request by cleishm**
And remove need to inject some dependencies when unrequired

### CODE REVIEW

**Comment by cleishm** [comment]
@davidegrohmann some suggested cleanups after the merge of #4719

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java [hunk 1]
```diff
@@ -129,6 +129,10 @@
     public static final Setting<Long> transaction_start_timeout =
             setting( "transaction_start_timeout", DURATION, "1s" );
 
+    @Description("The location of the internal diagnostics log.")
+    @Internal
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:133]: Current setting is marked with `@Internal`  and also contains _internal_ as part of the name. In the same time already closest neighbour `store_internal_log_rotation_threshold` do not follow this pattern. Let's mark all of them that matched to the principle?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/logging/StoreLogService.java [hunk 2]
```diff
@@ -42,51 +41,96 @@
...
     {
-        this( userLogProvider, fileSystem, storeDir, config, internalLogRotationThreshold, internalLogRotationDelay, maxInternalLogArchives,
-                jobScheduler.executor( JobScheduler.Groups.internalLogRotation ), rotationListener );
+        return new Builder().withRotation( internalLogRotationThreshold, internalLogRotationDelay, maxInternalLogArchives, jobScheduler );
     }
 
-    public StoreLogService( LogProvider userLogProvider, FileSystemAbstraction fileSystem, File storeDir, Config config,
-                            long internalLogRotationThreshold, int internalLogRotationDelay, int maxInternalLogArchives,
-                            Executor rotationExecutor, final Consumer<LogProvider> rotationListener ) throws IOException
+    public static StoreLogService inStoreDirectory( FileSystemAbstraction fileSystem, File storeDir ) throws IOException
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/logging/StoreLogService.java:106]: As result in new `StoreLogService` we will have Builder which provides 4 methods that we can use to configure new logService, static factory method `inStoreDirectory` that will provide constructed logService and 2 other static methods that will provide storeLogService builders. Can we simplify that and provide only one way of creating service or builder or factory methods that will return logService?

Comment (cleishm) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/logging/StoreLogService.java:106]: We could, except that the usages are much cleaner if there are methods that allow direct construction. E.g. https://github.com/neo4j/neo4j/pull/4767/files#diff-91905dbcabbe54b1fe19da086bac4b78R309, https://github.com/neo4j/neo4j/pull/4767/files#diff-3a67f986ab26ffad5395b2dd1793522fR180, etc

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/logging/StoreLogService.java:106]: that is a good example: we need StoreLogService to be constructed as fast and as clean as possible so yes we need to keep factory methods that result in logService itself, but not methods that will return builder. 
API users will have ability use or builder or directly exposed factory methods that will return keep things more clean and short.
And that i suppose we can apply to all new builders that coming. What do you think?

**Comment by davidegrohmann** [comment]
@cleishm @MishaDemianenko I like the changes. Perhaps there are further improvements to do, but overall it is good enough. Let's take of those improvements in a separate PR.

**Pull request merged**